### DISCUSSION TITLE
Build Relay-compliant graphql schema

### DISCUSSION THREAD

**Discussion by Andrej-Peplinski** [smallrye/smallrye-graphql#2159] [Q&A] [answered]
After search the documentation it is not clear to me if I can use smallrye-graphql to build a [relay-compliant](https://relay.dev/docs/guides/graphql-server-specification/) graphql server. So could you please tell me if that is possible and how?

**Comment by t1** [comment]
AFAIU, relay is a JavaScript GraphQL client lib. After my quick look, I didn't see any special GraphQL features it relies on. It should be straight forward to use smallrye-graphql to produce any GraphQL queries and mutations that you want to consume.

If you have more specific questions, please ask them!

**Reply by Andrej-Peplinski** [reply]
Thanks Rüdiger, for the quick reply but Relay is not "just" a JavaScript GraphQL client library. It's a spec on how to design your GraphQL schema so that the javascript client side can efficiently fetch (and cache) it's data. Other GraphQL libaries provide build-in support for Relay like for example [graphql-java](https://www.graphql-java-kickstart.com/tools/relay/) and [hotchocolate](https://chillicream.com/docs/hotchocolate/v13/defining-a-schema/relay). So, I wonder if it principle it is possible to use smallrye-graphql to build a Relay compliant schema, if anybody has done so and if so how?

**Reply by jmartisk** [reply, marked as answer]
Sorry I haven't tried that. But if you try, please share any findings, some best practices might be useful for the documentation.
Or, given the graphql-java support, we may try to figure out if that can be integrated on our side

**Reply by Andrej-Peplinski** [reply]
For anybody else that might run into the same challenge: I solved the Relay challenge by using the [graphql-java](https://www.graphql-java.com/documentation/relay) library directly - following the approach outlined [here](https://medium.com/@benjamin.habegger/step-by-step-guide-to-build-a-graphql-server-in-java-over-multiple-data-sources-30c0c7b4b409).
